pennylane/commutation_dag.py

```diff
@@ -0,0 +1,509 @@
+# Copyright 2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the CommutationDAG class which is used to generate a DAG (directed acyclic graph)
+representation of a quantum circuit from an Operator queue.
+"""
+
+from collections import OrderedDict
+import heapq
+import networkx as nx
+import pennylane as qml
+
+position = OrderedDict(
+    {
+        "Hadamard": 0,
+        "PauliX": 1,
+        "PauliY": 2,
+        "PauliZ": 3,
+        "SWAP": 4,
+        "ctrl": 5,
+    }
+)
+
+commutation_map = OrderedDict(
+    {
+        "Hadamard": [
+            1,
+            0,
+            0,
+            0,
+            0,
+            0,
+        ],
+        "PauliX": [
+            0,
+            1,
+            0,
+            0,
+            0,
+            0,
+        ],
+        "PauliY": [
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+        ],
+        "PauliZ": [
+            0,
+            0,
+            0,
+            1,
+            0,
+            1,
+        ],
+        "SWAP": [
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+        ],
+        "ctrl": [
+            0,
+            0,
+            0,
+            1,
+            0,
+            1,
+        ],
+    }
+)
+
+
+def intersection(wires1, wires2):
+    r"""Check if two operations are commuting
+
+    Args:
+        wires1 (pennylane.wires.Wires): First set of wires.
+        wires2 (pennylane.wires.Wires: Second set of wires.
+
+    Returns:
+         bool: True if the two sets of wires are not disjoint and False if disjoint.
+    """
+    if len(qml.wires.Wires.shared_wires([wires1, wires2])) == 0:
+        return False
+    return True
+
+
+def is_commuting(operation1, operation2):
+    r"""Check if two operations are commuting
```

Comment:
```suggestion
    r"""Check if two operations are commuting.
```

Reply:
@rmoyard might be good to add more to this docstring 🙂 Perhaps info such as:

- What method is used to determine commutation?
- Are there any restrictions?
- Which operations are supported?

Reply:
Yes I will add more information!